AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -696,58 +834,148 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
+        /// <returns>The PnL account update event from TWS.</returns>
+        public Task<PnLEventArgs> RequestPnL(
+            string accountCode,
+            string modelCode)
+        {
+            // Set the operation to cancel after 5 seconds
+            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
+            return this.RequestPnL(accountCode, modelCode, tokenSource.Token);
+        }
+
+        /// <summary>
+        /// Get the PnL of the account.
+        /// </summary>
+        /// <param name="accountCode">The account code</param>
+        /// <param name="modelCode">The model code</param>
         /// <param name="cancellationToken">The cancellation token used to cancel the request</param>
-        /// <returns>The security definitions for options</returns>
-        public Task<List<SecurityDefinitionOptionParameterEventArgs>> RequestSecurityDefinitionOptionParameters(
-            string underlyingSymbol,
-            string exchange,
-            string underlyingSecType,
-            int underlyingConId,
+        /// <returns>The PnL account update event from TWS.</returns>
+        public Task<PnLEventArgs> RequestPnL(
+            string accountCode,
+            string modelCode,
             CancellationToken cancellationToken)
         {
-            var taskSource = new TaskCompletionSource<List<SecurityDefinitionOptionParameterEventArgs>>();
+            PnLEventArgs pnlUpdateEvent = null;
+            var taskSource = new TaskCompletionSource<PnLEventArgs>();
+            EventHandler<PnLEventArgs> pnlUpdateEventHandler = null;
 
-            List<SecurityDefinitionOptionParameterEventArgs> securityDefinitionEvents = new List<SecurityDefinitionOptionParameterEventArgs>();
+            pnlUpdateEventHandler = (sender, args) =>
+            {
+                pnlUpdateEvent = args;
+                taskSource.TrySetResult(pnlUpdateEvent);
+            };
 
-            EventHandler<SecurityDefinitionOptionParameterEventArgs> securityDefinitionEventHandler = null;
-            EventHandler securityDefintionEndEventHandler = null;
+            this.twsCallbackHandler.PnLEvent += pnlUpdateEventHandler;
 
-            securityDefinitionEventHandler = (sender, args) =>
+            cancellationToken.Register(() =>
             {
-                securityDefinitionEvents.Add(args);
-            };
+                taskSource.TrySetCanceled();
+            });
 
-            securityDefintionEndEventHandler = (sender, args) =>
+            int requestId = this.twsRequestIdGenerator.GetNextRequestId();
+            this.clientSocket.RequestPnL(requestId, accountCode, modelCode);
+
+            return taskSource.Task;
+        }
+
+        /// <summary>
+        /// Request PnL update cancelation
+        /// </summary>
+        /// <param name="reqId">The request Id</param>
+        public void CancelPnL(int reqId)
+        {
+            this.clientSocket.CancelPnL(reqId);
```

Comment:
use `this.twsRequestIdGenerator.GetNextRequestId()` to generate the `reqId` and remove it from the parameter

Reply:
Following the documentation https://interactivebrokers.github.io/tws-api/pnl.html#portfolio_window_pnl it seems we need to pass the same reqId as the one provided when requesting the PnL.